website/docs/r/monitor_data_collection_rule_association.html.markdown

```diff
@@ -0,0 +1,153 @@
+
+Manages a Data Collection Rule Association.
+
+## Example Usage
+
+```hcl
+resource "azurerm_resource_group" "example" {
+  name     = "example-resources"
+  location = "West Europe"
+}
+
+resource "azurerm_virtual_network" "example" {
+  name                = "virtualnetwork"
+  address_space       = ["10.0.0.0/16"]
+  location            = azurerm_resource_group.example.location
+  resource_group_name = azurerm_resource_group.example.name
+}
+
+resource "azurerm_subnet" "example" {
+  name                 = "subnet"
+  resource_group_name  = azurerm_resource_group.example.name
+  virtual_network_name = azurerm_virtual_network.example.name
+  address_prefixes     = ["10.0.2.0/24"]
+}
+
+resource "azurerm_network_interface" "example" {
+  name                = "nic"
+  location            = azurerm_resource_group.example.location
+  resource_group_name = azurerm_resource_group.example.name
+
+  ip_configuration {
+    name                          = "internal"
+    subnet_id                     = azurerm_subnet.example.id
+    private_ip_address_allocation = "Dynamic"
+  }
+}
+
+resource "azurerm_linux_virtual_machine" "example" {
+  name                = "machine"
+  resource_group_name = azurerm_resource_group.example.name
+  location            = azurerm_resource_group.example.location
+  size                = "Standard_B1ls"
+  admin_username      = "adminuser"
+  network_interface_ids = [
+    azurerm_network_interface.example.id,
+  ]
+
+  admin_password = "example-Password@7890"
+
+  disable_password_authentication = false
+
+  os_disk {
+    caching              = "ReadWrite"
+    storage_account_type = "Standard_LRS"
+  }
+
+  source_image_reference {
+    publisher = "Canonical"
+    offer     = "UbuntuServer"
+    sku       = "16.04-LTS"
+    version   = "latest"
+  }
+}
+
+resource "azurerm_monitor_data_collection_rule" "example" {
+  name                = "example-dcr"
+  resource_group_name = azurerm_resource_group.example.name
+  location            = azurerm_resource_group.example.location
+  destinations {
+    azure_monitor_metrics {
+      name = "example-destination-metrics"
+    }
+  }
+  data_flow {
+    streams      = ["Microsoft-InsightsMetrics"]
+    destinations = ["example-destination-metrics"]
+  }
+}
+
+resource "azurerm_monitor_data_collection_endpoint" "test" {
+  name                = "example-dce"
+  resource_group_name = azurerm_resource_group.example.name
+  location            = azurerm_resource_group.example.location
+}
+
+# associate to a Data Collection Rule
+resource "azurerm_data_collection_rule_association" "example1" {
+  name                    = "example1-dcra"
+  target_resource_id      = azurerm_linux_virtual_machine.example.id
+  data_collection_rule_id = azurerm_monitor_data_collection_rule.example.id
+  description             = "example"
+}
+
+# associate to a Data Collection Endpoint
+resource "azurerm_data_collection_rule_association" "example1" {
```

Comment:
should this be named example2? I tried pasting the code from https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/monitor_data_collection_rule_association and hit a 'example1' already exists error from the resource above defined on line 96

Reply:
also line 106, `data_collection_endpoint_id = azurerm_monitor_data_collection_endpoint.test.id` ?

Reply:
Thanks for pointing this out, I have submitted another PR #18536 to fix this.